### ISSUE TITLE
Updating resource_labels field on google_container_cluster fails with a 400 error

### ISSUE THREAD

**Issue by chrisob** [terraform-providers/terraform-provider-google#5608]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version
```
Terraform v0.12.20
+ provider.google v3.7.0
```

### Affected Resource(s)

* google_container_cluster

### Terraform Configuration Files

```tf
resource "google_container_cluster" "default" {
  ...

  # First create a cluster without resource_labels set, then re-apply after setting a label 
  resource_labels = {
    key = "value"
  }
}
```

### Debug Output

https://gist.github.com/chrisob/52250906dc38c6f300673033a60c26a1

### Expected Behavior

TF should be able to update a GKE cluster's `resource_labels` on the fly.

### Actual Behavior

Because the request body doesn't include the field `labelFingerprint`, TF gets a 400 error back from GCP complaining that:

```
Error: googleapi: Error 400: Labels could not be set due to fingerprint mismatch; try again with the proper fingerprint: incorrect fingerprint supplied was "", the expected one is "a9dc16a7"., failedPrecondition
```

### Steps to Reproduce

1. `terraform apply` to create a `google_container_cluster` without any `resource_labels` set.
1. Add `resource_labels` to the `google_container_cluster`, and attempt another `terraform apply`

### Important Factoids

[Looking at the API spec for `setResourceLabels`](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters/setResourceLabels), it's pretty clear that the issue is the missing `labelFingerprint` field in the request body.

### References

Relevant GCP API call: https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters/setResourceLabels

* #0000

**Comment by edwardmedia** [comment]
@c2thorn I see the same problem. Debug.log is the same (above). In the plan I do see 
```
      ~ resource_labels             = {
          + "key" = "value"
        }
```

**Comment by eadred** [comment]
The same issue affects terraform-provider-google-beta (v2.18.1 in my case).

**Comment by c2thorn** [comment]
Thanks for contributing @eadred! Merged in your changes. If you ever want to contribute in the future, the easiest way is to make a change in our Magic Modules repo like so: https://github.com/GoogleCloudPlatform/magic-modules/pull/3128. That way it will make the same change to both providers.

Either way, thanks a ton for helping out!

**Comment by bappr** [comment]
Hey guys,
Is this issue supposed to be fixed already?
I am stucked because of that atm. Any advices?

**Comment by eadred** [comment]
The fixes didn't make it into v3.8.0 - presumably the next release will have them.

**Comment by c2thorn** [comment]
v3.9.0 was cut before the change went in. The fix should be available in v3.10.0 which should be released around 2/24. 
If needed sooner, you can always build direclty from the master branch.

**Comment by bwmaas** [comment]
Any chance we can get a v2.20+ that includes this fix?

**Comment by eadred** [comment]
The fix only went into HEAD, but you should be able to fork and cherry pick the fix onto v2.20. It worked OK without any changes on v2.18.
